### PR TITLE
Colm/ssttp 1151/change enter your bank details page

### PR DESCRIPTION
**Pull request by cavster**
[https://jira.tools.tax.service.gov.uk/browse/SSTTP-1151](https://jira.tools.tax.service.gov.uk/browse/SSTTP-1151)

### CODE REVIEW

File: app/uk/gov/hmrc/selfservicetimetopay/controllers/DirectDebitController.scala [hunk 1]
```diff
@@ -84,23 +86,33 @@ class DirectDebitController @Inject() (val messagesApi: play.api.i18n.MessagesAp
...
+                ))
             }
+          }
         )
       }
   }
 
-  private def checkBankDetails(bankDetails: BankDetails,accName:String)(implicit hc: HeaderCarrier) ={
+  def filterSortCodeErrors(form: Form[ArrangementDirectDebit]): Form[ArrangementDirectDebit] = {
+    val errorsNoDubs = form.errors.foldLeft[Seq[FormError]](Nil) { (acc, r) => {
```
Comment (nkaraolis) [inline app/uk/gov/hmrc/selfservicetimetopay/controllers/DirectDebitController.scala]: Can we rename this to something a bit clearer?

File: app/uk/gov/hmrc/selfservicetimetopay/controllers/DirectDebitController.scala [hunk 2]
```diff
@@ -84,23 +86,33 @@ class DirectDebitController @Inject() (val messagesApi: play.api.i18n.MessagesAp
...
-                    submission.schedule.get, directDebitFormWithBankAccountError.copy(data = Map("accountName" -> validFormData.accountName,
-                    "accountNumber" -> validFormData.accountNumber,"sortCode" -> validFormData.sortCode)),
-                      isBankError = true)
-              ))
+              case Some(bankDetails) => checkBankDetails(bankDetails, validFormData.accountName)
+              case None =>
+                val (sc1 :: sc2 :: sc3 :: _) = validFormData.sortCode.grouped(2).toList
+                Future.successful(BadRequest(direct_debit_form(submission.calculatorData.debits,
+                  submission.schedule.get, directDebitFormWithBankAccountError.copy(data = Map("accountName" -> validFormData.accountName,
+                    "accountNumber" -> validFormData.accountNumber, "sortCode1" -> sc1, "sortCode2" -> sc1, "sortCode3" -> sc1)),
```
Comment (nkaraolis) [inline app/uk/gov/hmrc/selfservicetimetopay/controllers/DirectDebitController.scala]: This copies the value from the first box into every box. I guess you want to use sc2 and sc3 here for the other boxes?

**Review by nkaraolis — CHANGES_REQUESTED** [review]
See comments:
Also, the errors for sort code and account number need to only have one error rather than both when a user doesn't input any data. Need to confirm this with design team though

File: app/uk/gov/hmrc/selfservicetimetopay/controllers/DirectDebitController.scala (hunk 1 above)
Comment (cavster) [reply app/uk/gov/hmrc/selfservicetimetopay/controllers/DirectDebitController.scala]: remove duplicated errors?

Comment (nkaraolis) [reply app/uk/gov/hmrc/selfservicetimetopay/controllers/DirectDebitController.scala]: Something like that should be okay

**Comment by nkaraolis** [comment]
Correct values are now being put back in however, the error messages are slightly off.
If there is no data inputted in either the sort code and account number fields only need to show once message which should be depending on the form field:
Enter a sort code or Enter an account number

**Comment by nkaraolis** [comment]
Everything looks good now, just optimise the imports when you resolve the conflicts :D

**Review by nkaraolis — APPROVED** [review]

**Pull request merged**